lib/abstract_client.js

```diff
@@ -0,0 +1,88 @@
+'use strict';
+
+const Beanstalk = require('fivebeans').client;
+const moment = require('moment');
+const bb = require('bluebird');
+const _ = require('lodash');
+
+/**
+ * Abstract beanstalk client class
+ */
+class AbstractClient {
+	constructor(config) {
+		this.client = null;
+		this.connected = false;
+		this.host = config.host?config.host : '127.0.0.1';
+		this.port = config.port?config.port : 11300;
+		this.tube = config.tube?config.tube : 'default';
+	}
+
+	async start() {
+		const _this = this;
+		if (!_this.connected) {
+			return new Promise(resolve => {
```

Comment:
It will never resolve/reject when error?

Reply:
may consider referencing `ioredis` for the connection/offline queue handling?

https://github.com/luin/ioredis/blob/master/lib/redis.js#L249-L318